src/client/components/filter-menu/string-filter-menu/string-filter-menu.tsx

```diff
@@ -110,73 +85,44 @@ export class StringFilterMenu extends React.Component<StringFilterMenuProps, Str
     return filterOptions;
   }
 
-  renderMenuControls() {
-    const { filterMode, searchText } = this.state;
-
-    return <div className="string-filter-menu-controls">
-      <div className="side-by-side">
-        <FilterOptionsDropdown
-          selectedOption={filterMode}
-          onSelectOption={this.onSelectFilterOption}
-          filterOptions={this.getFilterOptions()}
-        />
-        <div className="search-box">
-          <ClearableInput
-            placeholder="Search"
-            focusOnMount={true}
-            value={searchText}
-            onChange={this.updateSearchText}
-          />
-        </div>
-      </div>
-    </div>;
+
+  renderFilterControls(): JSX.Element {
+    const { dimension, clicker, essence, timekeeper, onClose } = this.props;
+    const { filterMode } = this.state;
+    const onClauseChange = this.updateFilter;
+    const props = { dimension, clicker, essence, timekeeper, onClose, onClauseChange };
```

Comment:
why don't you add `filterMode` right here? then you could switch used component only.

Reply:
TSX typing issues. If I add this there, `props` will have `filterMode: FilterMode`, and switch wouldn't narrow it's type inside object so types on component's won't match. If I narrow type via switch and then add to object, it has correct type.